crates/router/src/route.rs

```diff
@@ -118,54 +118,61 @@ impl Route {
     /// - /a/[id].js
     /// - /[id]/b.wasm
     /// - /[id]/[other].wasm
+    /// - /[id]/[..all].wasm
     ///
     /// We need to establish a priority. The lower of the returned number,
     /// the more priority it has. This number is calculated based on the number of used
     /// parameters, as fixed routes has more priority than parameted ones.
     ///
     /// To avoid collisions like `[id]/b.wasm` vs `/a/[id].js`. Every depth level will
     /// add an extra +1 to the score. So, in case of `[id]/b.wasm` vs `/a/[id].js`,
-    /// the /a/b path will be managed by `[id]/b.wasm`
+    /// the /a/b path will be managed by `/a/[id].js`
     ///
     /// In case it cannot manage it, it will return -1
     pub fn affinity(&self, url_path: &str) -> RouteAffinity {
         let mut score: i32 = 0;
         let mut split_path = self.path.split('/').peekable();
 
-        for (depth, portion) in url_path.split('/').enumerate() {
+        for (depth, segment) in url_path.split('/').enumerate() {
             match split_path.next() {
-                Some(el) if el == portion => continue,
-                Some(el) if PARAMETER_REGEX.is_match(el) => {
-                    score += depth as i32;
-                    continue;
-                }
-                _ => return RouteAffinity::CannotManage,
+                None => return RouteAffinity::CannotManage,
+                Some(el) if el == segment => continue,
+                Some(el) => match PARAMETER_REGEX.captures(el) {
+                    None => return RouteAffinity::CannotManage,
+                    Some(caps) => match (caps.name("ellipsis"), caps.name("segment")) {
+                        (Some(_), Some(_)) => return RouteAffinity::CanManage(i32::MAX),
```

Comment:
Here, I'm wondering the possible collisions when there're multiple "catch all" routes. For example, let's assume we have these files:

1. `[id].wasm`
2. `fixed.wasm`
1. `./[...all].wasm`
1. `./sub/[...all].wasm`

Now, let's assume the following requests:

| Path | `[id].wasm` | `fixed.wasm` | `./[...all].wasm` | `./sub/[...all].wasm` |
| --- | --- | --- | --- | --- |
| `/test` | Can(1) | Cannot | Can(i32::MAX) | Cannot | 
| `/fixed | Can(1) | Can(0) | Can(i32::MAX) | Cannot |
| `/sub/test` | Cannot | Cannot | Can(i32::MAX) | Can(i32::MAX) |

Since 3 and 4 are "catch all" routes, I would suggest that 4 manages `/sub/test`. To enable this behavior, I would suggest to subtract the `depth` to `i32::MAX`. In that way, `wws` will give higher priority to more specific "catch all" routes.

Wdyt?

```suggestion
                        (Some(_), Some(_)) => return RouteAffinity::CanManage(i32::MAX - depth),
```